apps/client/app/components/SupplierInvoiceForm.tsx

```diff
@@ -0,0 +1,155 @@
+'use client';
+
+import React, { useState } from 'react';
+import { useForm, Controller, useFieldArray } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import { SupplierInvoice, Item, PurchaseOrder } from '../interfaces';
+
+interface SupplierInvoiceFormProps {
+  supplierInvoice?: SupplierInvoice;
+  items: Item[];
+  purchaseOrders: PurchaseOrder[];
+  mode: 'create' | 'edit';
```

Comment:
these are repeated often, would. be. good to create once and import where needed

Reply:
Great point. I refactored the code to use an enum